src/iterateJsdoc.js

```diff
@@ -962,6 +963,14 @@ const iterate = (
     return;
   }
 
+  if (
+    context.options[0] &&
```

Comment:
I think we could also support a check for `settings.publicOnly` here in case the `publicOnly` setting is used in place of an option.